File: OCH_Tao/ABC/B/299.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, T] = input[0].split(" ").map(x => parseInt(x, 10));
const C = input[1].split(" ").map(x => parseInt(x, 10));
const R = input[2].split(" ").map(x => parseInt(x, 10));
const X = C.includes(T) ? T : C[0];
let L = [];
for (let i = 0; i < N; i++) {
  if (C[i] === X) {
    L.push([R[i], i + 1]);
  }
}
L.sort((a, b) => b[0] - a[0]);
console.log(L[0][1]);

File: OCH_Tao/ABC/B/299.py
N, T = map(int, input().split())
C = list(map(int, input().split()))
R = list(map(int, input().split()))
X = T if T in set(C) else C[0]
L = []
for i in range(N):
  if C[i] == X:
    L.append([R[i], i+1])
print(sorted(L, key=lambda x: x[0], reverse=True)[0][1])
